utils/EventRepository.js

```diff
@@ -227,12 +230,30 @@ class EventRepository {
 
     // modify the url for event specific endpoints
     const url = this.url.slice(-1) == "/" ? this.url.slice(0, -1) : this.url;
-    // add items to the queue
-    this.payloadQueue.addItem({
-      url: `${url}/v1/${type}`,
-      headers,
-      message,
-    });
+    if (this.useBeacon) {
+      const targetUrl = `${url}/beacon/v1/batch`;
+      beaconQueue.enqueue(targetUrl, headers, message, this.writeKey);
+    } else {
+      // add items to the queue
+      this.payloadQueue.addItem({
+        url: `${url}/v1/${type}`,
+        headers,
+        message,
+      });
+    }
+  }
+
+  sendQueueDataForBeacon() {
+    this.beaconQueue.sendDataFromQueueAndDestroyQueue();
+  }
+
+  initializeTransportMechanism() {
+    const sendQueueData = this.sendQueueDataForBeacon.bind(this);
+    const url = this.url.slice(-1) == "/" ? this.url.slice(0, -1) : this.url;
+    const targetUrl = `${url}/beacon/v1/batch`;
+    this.beaconQueue.url = targetUrl;
+    this.beaconQueue.writekey = this.writeKey;
+    window.addEventListener("unload", sendQueueData);
   }
```

Comment:
Why do these have to be here than in the beacon lib?

Reply:
Yes can be moved to beacon lib.

Reply:
done